main.cpp

```diff
@@ -90,6 +91,115 @@ good to go!
 
  Wait for my code review.
  */
+struct FloatType
+{
+   
+    float add(float lhs,float rhs);
+    float subtract(float lhs, float rhs);
+    float multiply(float lhs, float rhs);
+    float divide(float lhs, float rhs); 
+};
+
+struct DoubleType
+{
+    double add(double lhs, double rhs);
+    double subtract(double lhs, double rhs);
+    double multiply(double lhs, double rhs);
+    double divide(double lhs, double rhs);
+};
+
+struct IntType
+{
+    int add(int lhs, int rhs);
+    int subtract(int lhs, int rhs);
+    int multiply(int lhs, int rhs);
+    int divide(int lhs, int rhs);
+};
+
+float FloatType::add(float lhs, float rhs)
+{
+    return lhs + rhs;
+}
+
+float FloatType::subtract(float lhs, float rhs)
+{
+    return lhs - rhs;
+}
+
+float FloatType::multiply(float lhs, float rhs)
+{
+    return lhs * rhs;
+}
+float FloatType::divide(float lhs, float rhs)
+{
+    if (rhs == 0.0f)
+    {
+        std::cout << "warning, floating point division by zero returns 'inf' !" << std::endl;
+        return lhs / rhs;
+    }
+    else 
+    {
+        return lhs / rhs;
+    
+    }
+}
+
+double DoubleType::add(double lhs, double rhs)
+{
+    return lhs + rhs;
+}
+
+double DoubleType::subtract(double lhs, double rhs)
+{
+    return lhs - rhs;
+
+}
+
+double DoubleType::multiply(double lhs, double rhs)
+{
+    return lhs * rhs;
+}
+
+double DoubleType::divide(double lhs, double rhs)
+{
+    if(rhs == 0.0)
+    {
+        std::cout << "warning, floating point division by zero returns 'inf' !" << std::endl;
+        return lhs / rhs;
+    }
+    else
+    {
+        return lhs / rhs;
+    }
+}
```

Comment:
Don't Repeat Yourself
The only time you issue the warning is when `rhs` is `0.0` 
You always return the computation. 
There is no need for 2 return statements.
Revise the logic.
- [ ] revise